lld/ELF/Driver.cpp

```diff
@@ -566,6 +570,7 @@ constexpr const char *knownZFlags[] = {
     "origin",
     "pac-plt",
     "pack-relative-relocs",
+    "pack-relative-auth-relocs",
```

Comment:
Is there justification for adding a new option?
Why can't we just use `pack-relative-relocs`?
This is a new ABI and we are adding stuff that existing linkers don't know, so there isn't any compatibility concern.

If there is sufficient justification for a new option, we should have agreement with binutils folks on the name.

Reply:
Hmm, since we place auth packed relocs to a separate section other than usual relr section, this should not cause compatibility issues.
It looks like it would be better use `--pack-dyn-relocs` then, like it's already done for android packed relocs. They don't rely on `-z pack-relative-relocs`.

```
static std::pair<bool, bool> getPackDynRelocs(opt::InputArgList &args) {
  StringRef s = args.getLastArgValue(OPT_pack_dyn_relocs, "none");
  if (s == "android")
    return {true, false};
  if (s == "relr")
    return {false, true};
  if (s == "android+relr")
    return {true, true};

  if (s != "none")
    error("unknown --pack-dyn-relocs format: " + s);
  return {false, false};
}
```

Reply:
Update on my previous comment: changing `getPackDynRelocs` as I suggested above is probably not the best option since android packed relocs just replace the regular relocation section (`lld::elf::Partition::relaDyn`), while relr relocs (both regular and auth ones) have their own separate sections (`lld::elf::Partition::relrDyn` and `lld::elf::Partition::relrAuthDyn`). So, changed the logic to what you have suggested - just use `pack-relative-relocs` option for both regular and auth relr relocations. See d3411595fbbe37a14b0187ff0d134bd105892931.

Reply:
The extra option has been removed. Resolved